src/olympia/devhub/utils.py

```diff
@@ -307,3 +312,48 @@ def add_dynamic_theme_tag(version):
     files = version.all_files
     if any('theme' in file_.webext_permissions_list for file_ in files):
         Tag(tag_text='dynamic theme').save_tag(version.addon)
+
+
+def get_addon_akismet_reports(user, user_agent, referrer, upload=None,
+                              addon=None, data=None):
+    if not waffle.switch_is_active('akismet-spam-check'):
+        return []
+    assert addon or upload
+    properties = ('name', 'summary', 'description')
+
+    if upload:
+        addon = addon or upload.addon
+        data = data or Addon.resolve_webext_translations(
+            parse_addon(upload, addon, user), upload)
+
+    if not data:
+        return []  # bail early if no data to skip Translation lookups
+    if addon and addon.has_listed_versions():
+        translation_ids = (
+            getattr(addon, prop + '_id', -1) for prop in properties)
+        # Just get all the values together to make it simplier
+        existing_data = {
+            text_type(value)
+            for value in Translation.objects.filter(id__in=translation_ids)}
+    else:
+        existing_data = ()
+
+    reports = []
+    for prop in properties:
+        locales = data.get(prop)
+        if not locales:
+            continue
+        if isinstance(locales, dict):
+            # Avoid spam checking the same value more than once by using a set.
+            locale_values = set(locales.values())
+        else:
+            # It's not a localized dict, it's a flat string; wrap it anyway.
+            locale_values = {locales}
+        for comment in locale_values:
+            if not comment or comment in existing_data:
+                # We don't want to submit empty or unchanged content
+                continue
+            reports.append(AkismetReport.create_for_addon(
+                upload, addon, user, prop, comment, user_agent,
+                referrer))
```

Comment:
This list of arguments is so long that it deserves explicit assignments, e.g `.create_for_addon(upload=upload, addon=addon` (etc) which is much less error prone.